### ISSUE TITLE
A bunch of unit test failures

### ISSUE THREAD

**Issue by BrettDong** [CleverRaven/Cataclysm-DDA#19560]
```
Loading content (dda)
Finalizing and verifying

~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
cata_test is a Catch v1.5.8 host application.
Run with -? for options

-------------------------------------------------------------------------------
Zombie attacking a character
  10/10/10/10, 8 all skills, survivor suit
      Then: The chance to hit is between 1.25% and 3.75%
-------------------------------------------------------------------------------
melee_test.cpp:124
...............................................................................

melee_test.cpp:62: FAILED:
  REQUIRE( prob < high )
with expansion:
  0.0383f < 0.0375f
with messages:
  Zombie has get_hit() == 4.000000
  Has get_dodge() == 13.000000

-------------------------------------------------------------------------------
monster_speed_trig
-------------------------------------------------------------------------------
monster_test.cpp:366
...............................................................................

monster_test.cpp:274: FAILED:
  CHECK( stat_pair.second.avg() == Approx(100.0).epsilon(0.1) )
with expansion:
  114.93333f == Approx( 100.0 )
with message:
  Monster:mon_zombie_dog Dist: 30 moves: 114.933

monster_test.cpp:279: FAILED:
  CHECK( stat_pair.second.avg() == Approx(100.0).epsilon(0.03) )
with expansion:
  110.28542f == Approx( 100.0 )
with message:
  Monster:mon_zombie_dog Slope: 0 moves: 110.285 types: 96

monster_test.cpp:289: FAILED:
  CHECK( stat_pair.second.avg() == Approx(100.0).epsilon(0.05) )
with expansion:
  160.64999f == Approx( 100.0 )
with message:
  Monster:mon_zombie_dog Angle: -294 moves: 160.65 types: 4 samples: 101, 219,
  330, 96, 375, 200, 107, 131, 172, 226, 102, 155, 202, 111, 111, 142, 97, 122,
  114, 100, 

monster_test.cpp:289: FAILED:
  CHECK( stat_pair.second.avg() == Approx(100.0).epsilon(0.05) )
with expansion:
  235.0f == Approx( 100.0 )
with message:
  Monster:mon_zombie_dog Angle: -281 moves: 235 types: 4 samples: 843, 173, 174
  , 247, 113, 326, 198, 294, 267, 274, 236, 172, 181, 155, 178, 104, 189, 115,
  286, 175, 

monster_test.cpp:289: FAILED:
  CHECK( stat_pair.second.avg() == Approx(100.0).epsilon(0.05) )
with expansion:
  147.64999f == Approx( 100.0 )
with message:
  Monster:mon_zombie_dog Angle: -260 moves: 147.65 types: 4 samples: 101, 168,
  129, 115, 197, 97, 165, 130, 148, 171, 168, 143, 180, 154, 137, 186, 169, 125
  , 135, 135, 

-------------------------------------------------------------------------------
on_load-sane-values
  Awake for 10 minutes, gaining hunger/thirst/fatigue
-------------------------------------------------------------------------------
npc_test.cpp:41
...............................................................................

npc_test.cpp:55: FAILED:
  CHECK( test_npc.get_hunger() >= five_min_ticks - margin )
with expansion:
  0 >= 1

npc_test.cpp:56: FAILED:
  CHECK( test_npc.get_thirst() >= five_min_ticks - margin )
with expansion:
  0 >= 1

-------------------------------------------------------------------------------
on_load-sane-values
  Awake for 2 days, gaining hunger/thirst/fatigue
-------------------------------------------------------------------------------
npc_test.cpp:41
...............................................................................

npc_test.cpp:69: FAILED:
  CHECK( test_npc.get_hunger() >= five_min_ticks - margin )
with expansion:
  141 >= 566 (0x236)

npc_test.cpp:70: FAILED:
  CHECK( test_npc.get_thirst() >= five_min_ticks - margin )
with expansion:
  140 >= 566 (0x236)

-------------------------------------------------------------------------------
on_load-sane-values
  Sleeping for 6 hours, gaining hunger/thirst (not testing fatigue due to lack
  of effects processing)
-------------------------------------------------------------------------------
npc_test.cpp:41
...............................................................................

npc_test.cpp:93: FAILED:
  CHECK( test_npc.get_hunger() >= expected_change - margin )
with expansion:
  6 >= 26

npc_test.cpp:94: FAILED:
  CHECK( test_npc.get_thirst() >= expected_change - margin )
with expansion:
  10 >= 26

-------------------------------------------------------------------------------
Player body temperatures converge on expected values.
  Nude target temperatures.
-------------------------------------------------------------------------------
player_test.cpp:66
...............................................................................

player_test.cpp:44: FAILED:
  CHECK( high > p->temp_cur[0] )
with expansion:
  600 (0x258) > 1156 (0x484)

player_test.cpp:44: FAILED:
  CHECK( high > p->temp_cur[0] )
with expansion:
  2100 (0x834) > 2656 (0xa60)

player_test.cpp:44: FAILED:
  CHECK( high > p->temp_cur[0] )
with expansion:
  3600 (0xe10) > 4044 (0xfcc)

player_test.cpp:43: FAILED:
  CHECK( low < p->temp_cur[0] )
with expansion:
  4900 (0x1324) < 4677 (0x1245)

player_test.cpp:43: FAILED:
  CHECK( low < p->temp_cur[0] )
with expansion:
  6400 (0x1900) < 5455 (0x154f)

player_test.cpp:43: FAILED:
  CHECK( low < p->temp_cur[0] )
with expansion:
  7900 (0x1edc) < 6122 (0x17ea)

player_test.cpp:43: FAILED:
  CHECK( low < p->temp_cur[0] )
with expansion:
  9400 (0x24b8) < 6733 (0x1a4d)

-------------------------------------------------------------------------------
Player body temperatures converge on expected values.
  Lightly clothed target temperatures
-------------------------------------------------------------------------------
player_test.cpp:66
...............................................................................

player_test.cpp:44: FAILED:
  CHECK( high > p->temp_cur[0] )
with expansion:
  600 (0x258) > 1178 (0x49a)

player_test.cpp:44: FAILED:
  CHECK( high > p->temp_cur[0] )
with expansion:
  2100 (0x834) > 2674 (0xa72)

player_test.cpp:44: FAILED:
  CHECK( high > p->temp_cur[0] )
with expansion:
  3600 (0xe10) > 4035 (0xfc3)

player_test.cpp:43: FAILED:
  CHECK( low < p->temp_cur[0] )
with expansion:
  4900 (0x1324) < 4801 (0x12c1)

player_test.cpp:43: FAILED:
  CHECK( low < p->temp_cur[0] )
with expansion:
  6400 (0x1900) < 5595 (0x15db)

player_test.cpp:43: FAILED:
  CHECK( low < p->temp_cur[0] )
with expansion:
  7900 (0x1edc) < 6292 (0x1894)

player_test.cpp:43: FAILED:
  CHECK( low < p->temp_cur[0] )
with expansion:
  9400 (0x24b8) < 6836 (0x1ab4)

-------------------------------------------------------------------------------
Player body temperatures converge on expected values.
  Heavily clothed target temperatures
-------------------------------------------------------------------------------
player_test.cpp:66
...............................................................................

player_test.cpp:44: FAILED:
  CHECK( high > p->temp_cur[0] )
with expansion:
  600 (0x258) > 1156 (0x484)

player_test.cpp:44: FAILED:
  CHECK( high > p->temp_cur[0] )
with expansion:
  2100 (0x834) > 2531 (0x9e3)

player_test.cpp:44: FAILED:
  CHECK( high > p->temp_cur[0] )
with expansion:
  3600 (0xe10) > 4100 (0x1004)

player_test.cpp:44: FAILED:
  CHECK( high > p->temp_cur[0] )
with expansion:
  5100 (0x13ec) > 5400 (0x1518)

player_test.cpp:43: FAILED:
  CHECK( low < p->temp_cur[0] )
with expansion:
  6400 (0x1900) < 5923 (0x1723)

player_test.cpp:43: FAILED:
  CHECK( low < p->temp_cur[0] )
with expansion:
  7900 (0x1edc) < 6605 (0x19cd)

player_test.cpp:43: FAILED:
  CHECK( low < p->temp_cur[0] )
with expansion:
  9400 (0x24b8) < 7228 (0x1c3c)

-------------------------------------------------------------------------------
Player body temperatures converge on expected values.
  Arctic gear target temperatures
-------------------------------------------------------------------------------
player_test.cpp:66
...............................................................................

player_test.cpp:44: FAILED:
  CHECK( high > p->temp_cur[0] )
with expansion:
  600 (0x258) > 1134 (0x46e)

player_test.cpp:44: FAILED:
  CHECK( high > p->temp_cur[0] )
with expansion:
  2100 (0x834) > 2622 (0xa3e)

player_test.cpp:44: FAILED:
  CHECK( high > p->temp_cur[0] )
with expansion:
  3600 (0xe10) > 3975 (0xf87)

player_test.cpp:44: FAILED:
  CHECK( high > p->temp_cur[0] )
with expansion:
  5100 (0x13ec) > 5389 (0x150d)

player_test.cpp:44: FAILED:
  CHECK( high > p->temp_cur[0] )
with expansion:
  6600 (0x19c8) > 6850 (0x1ac2)

player_test.cpp:43: FAILED:
  CHECK( low < p->temp_cur[0] )
with expansion:
  7900 (0x1edc) < 7567 (0x1d8f)

player_test.cpp:43: FAILED:
  CHECK( low < p->temp_cur[0] )
with expansion:
  9400 (0x24b8) < 8416 (0x20e0)

-------------------------------------------------------------------------------
visitable_remove
     Given: An adjacent vehicle contains several bottles of water
-------------------------------------------------------------------------------
visitable_remove.cpp:22
...............................................................................

visitable_remove.cpp:412: FAILED:
  REQUIRE( count_items( sel, container_id ) == count )
with expansion:
  6 == 5

===============================================================================
test cases:      43 |      38 passed | 3 failed |  2 failed as expected
assertions: 1549689 | 1549648 passed | 7 failed | 34 failed as expected
```

**Comment by macrosblackd** [comment]
The question becomes, what PRs were pushed recently that changed melee code?

**Comment by mugling** [comment]
I think that's been broken for some time

**Comment by mutability** [comment]
Exclude [!mayfail] to exclude the expected-to-fail tests.

Melee tests were failing, but got fixed: #19021
on_load-sane-values failure here: #19022
temperature failures here: #19023 (this probably needs reopening as I bisected it)

Both the melee and the speed tests are, iirc, statistical tests, so possibly they just failed that one time. If they are repeatedly failing then it's an actual problem.

**Comment by mutability** [comment]
visitable_remove is a fairly old one that I missed bisecting at the time when I looked at the tests.

**Comment by codemime** [comment]
`on_load-sane-values` failure is resolved in #18471 (presumably forgotten).

**Comment by kevingranade** [comment]
All tests are either passing or excluded with "[.]", fixing them is always good, but we don't need an eternal issue to track them.

If you have a particular idea for fixing or adding a specific unit test though, please do post an issue, they're very valuable.
